rust/connlib/tunnel/src/lib.rs

```diff
@@ -134,6 +134,16 @@ where
     CB: Callbacks + 'static,
 {
     pub fn poll_next_event(&mut self, cx: &mut Context<'_>) -> Poll<Result<Event<ClientId>>> {
+        match self.role_state.poll(cx) {
+            Poll::Ready(ids) => {
+                cx.waker().wake_by_ref();
```

Comment:
I'm still learning this poll style. Does the wake call here mean, there might be more work to do after this function runs, and the runtime should poll it again as soon as it can?

Reply:
Exactly, since we handled the event internally and the poll didn't return `Pending` the waker is not scheduled to be woken again if we don't set it explicitly here.